core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java

```diff
@@ -55,6 +61,22 @@ private static ConfigDef configDef(
             ConfigDef.Importance.MEDIUM,
             CACHE_RETENTION_DOC
         );
+        configDef.define(
+            CACHE_FETCH_THREAD_POOL_SIZE_CONFIG,
+            ConfigDef.Type.INT,
+            0,
+            ConfigDef.Range.between(0, 1024),
```

Comment:
1024 looks quite big, should it be smaller?

Reply:
Yeah, didn't want it be to be too constrained 😅 ok to shrink it, do you have a better value in mind?

Reply:
Found that the limit on the ForkJoinPool is 32767
```
static final int MAX_CAP      = 0x7fff;        // max #workers - 1
```
so, maybe 1024 (or even higher?) may be ok?

Reply:
But that `0x7fff` is the biggest possible at all, correct? Assuming stack size to be ~1MB then this amount of threads will consume 32GB of memory if they will be allocated simultaneously, right?
In case of 1024 it will be additional gigabyte in the same situation as far as I understand. Are we ok with that?
Also did we do any performance testing with this high parallelism?

Reply:
Of course not :) It's just the upper-bound. The aim of these configs is just to tune, depending on the user's resources -- that's why I'm being relaxed with the upper-bound. 
Next PR is going to add metrics for the thread-pools, so it can be observed if thread-pools are the bottleneck, and users can tune accordingly.

Reply:
I'm just wondering how tuning this to any higher value affects the performance/memory consumption and if there is any implicit limit. For now this is fine I guess but would be interesting to see how different values will affect benchmarks.

Reply:
Fair point, as soon as metrics start to point into any of these thread pools will start with the perf tuning and share results. There's some discussion here: https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/issues/584#issuecomment-2337889015